### PR TITLE
setup and initial code 

### PR DESCRIPTION
**Pull request by erenasena**

### CODE REVIEW

File: LICENSE.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
```
Comment (princessamygdala) [inline LICENSE.md:19]: What I did in my license is to say that I also use other packages from python, which kind of have their own license (I copied this from somewhere). So if you are using some functions as well, I would add that :)

File: .ipynb_checkpoints/README-checkpoint.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Neural Networks 
+
+***'sensus' is the name of my package. It means 'perception' or 'sense' in Latin. It was the only unique package name I could
```
Comment (princessamygdala) [inline .ipynb_checkpoints/README-checkpoint.md:3]: i think its a beautiful name!

File: .ipynb_checkpoints/README-checkpoint.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Neural Networks 
+
+***'sensus' is the name of my package. It means 'perception' or 'sense' in Latin. It was the only unique package name I could 
+come up with that was both relatable to neural networks and was unique among Python packages. The code is in the package folder. ***
+
+*In this project, I want to build a neural network that can recognize handwritten digits, using Python. The functions involved in a simple neural network are clear and the training data are available online. For the functions to make sense, I wrote a brief description of how a neural network works. I should be able to finish this on time and I do not know what I can cut if I run out of time. If anything is left out from the structure I explained below, it will not work. If I have time left, I will build a Shiny app that shows the effect of drift in Brownian motion simulations.*
```
Comment (princessamygdala) [inline .ipynb_checkpoints/README-checkpoint.md:6]: sounds like an awesome project and I am excited to see how it works. The thing you want to add also seems interesting, however, does it have anything to do with the digit recognition package? If not, I would maybe think about adding something to the package you already have, for instance, not only recognizing English digits but also digits from other languages (but this is just an example suggestion of course)

File: .ipynb_checkpoints/README-checkpoint.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+The activations in one layer are propagated forward to the next layer as a weighted sum. That is, the input of a neuron in one layer is the weighted sum of the outputs (i.e., activations) of the neurons in the previous layer. In addition, each neuron has a bias. The bias of a neuron can be thought of as its ‘threshold’ – its tendency to be on or off. This bias is similar to the intercept in a regression equation. In effect, the activations of the neurons in the preceding layer, the weights and the biases combine to determine the activation of a neuron. This activation value goes into an activation function, which converts it into the output of that neuron. I will use the sigmoid function for simplicity, which compresses the activation value into a number between zero and one. 
+
```
Comment (princessamygdala) [inline .ipynb_checkpoints/README-checkpoint.md:13]: I find the description really interesting and wonderful. It is, however, quite abstract so you should think about if this is enough for your audience or if you want to give some more examples in what this network is then actually doing in the end and where you can apply it.

File: .ipynb_checkpoints/README-checkpoint.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+To summarize, these are the functions I will use in this project. 
+1. The activation function defines the nodes and their outputs
+2. The weights and the biases determine the inputs to subsequent layers 
+3. The loss function is the feedback mechanism
+4. Gradient descent optimizes learning from the feedback
```
Comment (princessamygdala) [inline .ipynb_checkpoints/README-checkpoint.md:20]: the last part could be a bit more elaborate.. :)

File: sensus/sensus/.ipynb_checkpoints/code-checkpoint.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+                          # b is a vector with the biases of the computed layer 
+        """ The input of the neuron """
+        z = w * a + b # z is the resulting vector
+        return z 
+    
+    def sigmoid(self, z):
+        """ The output of the neuron """
+        output = 1 / (1 + np.exp(-z)) # Numpy applies the sigmoid function elementwise
+        return output # this is the output vector, which will be the activation vector for the next layer
+
```
Comment (princessamygdala) [inline sensus/sensus/.ipynb_checkpoints/code-checkpoint.py:31]: nicely documented!

File: sensus/.ipynb_checkpoints/setup-checkpoint.py [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+
+setup(name='sensus',
+      version='0.1',
+      description='A library for building a simple neural network.',
+      url='https://github.com/Programming-The-Next-Step/Neural_network.git',
+      author='Eren Asena',
+      author_email='eren.asena@student.uva.nl',
+      license='MIT',
+      packages=['sensus'],
+      zip_safe=False)
```
Comment (princessamygdala) [inline sensus/.ipynb_checkpoints/setup-checkpoint.py]: that is cool, what is this and why is it necessary?

File: .ipynb_checkpoints/README-checkpoint.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+The network reaches the correct response by learning from its mistakes. The quality of a response is communicated back to the network via back propagation. A loss function is defined, such as the sum of the squared differences. This calculates the distance between the correct response and the predicted response. The derivative of the loss function with respect to the weights gives the change in the loss function based on the change in the weights. The aim is to find the weights and the biases that minimize the loss function. This is done by an algorithm called the gradient descent. The gradient of a function is the direction in which it increases the most steeply. The negative of the gradient gives us the steepest decrease. The neural network minimizes the loss function by iterating according to the gradient descent. One can picture sliding on the slope of a function until we reach the lowest point on the curve. 
+
+To summarize, these are the functions I will use in this project.
```
Comment (terHahaha) [inline .ipynb_checkpoints/README-checkpoint.md:16]: It's a great project. It seems you've given it a lot of thought and are very capable of completing it. I thought to utilise the sum of squared difference was very intuitive. I was just left a bit unsure on how you'd decided on a cut-off to deem a response correct although I'm sure you've got that figured out with the descent.

File: .ipynb_checkpoints/README-checkpoint.md (hunk 5 above)
Comment (terHahaha) [inline .ipynb_checkpoints/README-checkpoint.md:20]: Clear but I agree, could do with some elaboration especially regarding the time frame you might be looking at to complete it all.

File: sensus/sensus/code.py [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+        """ The input of the neuron """
+        z = w * a + b # z is the resulting vector
+        return z 
+    
+    def sigmoid(self, z):
+        """ The output of the neuron """
+        output = 1 / (1 + np.exp(-z)) # Numpy applies the sigmoid function elementwise
+        return output # this is the output vector, which will be the activation vector for the next layer
+    
+
```
Comment (terHahaha) [inline sensus/sensus/code.py:32]: It all follows very logically till now. Something that I always utilise is separating my classes from my main file, as I think it allows this to stay a bit clearer. From your description, I feel you have a lot of code coming. Though again, from this chunk it's a bit unclear if that is what you were doing to begin with.

**Pull request merged**